File: e2e/playwright-typescript/tests/create-task/create-task.axe.spec.ts
import { test } from '@/fixtures';
import * as allure from "allure-js-commons";
import { AllureEpic } from '@/data/AllureEpic';

test.beforeAll(async () => {
  await allure.epic(AllureEpic.ACCESSIBILITY);
  await allure.feature('Create task');
});

test.describe('Create task form - accessibility', () => {
  test('should have no accessibility violations on create task form', { tag: '@accessibility' }, async ({ step, validate }) => {
    // given
    await step.navigation.openMainPage();
    await step.tasks.openCreateTaskForm();

    // when
    const results = await step.accessibility.analyze();

    // then
    await validate.accessibility.hasNoViolations(results);
  });
});

File: e2e/capybara-ruby/spec/create_task/create_task_axe_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../data/allure_epic'

RSpec.describe 'Create task form - accessibility', type: :system, accessibility: true,
                                                    epic: AllureEpic::ACCESSIBILITY,
                                                    feature: 'Create task',
                                                    tms: '100' do
  it 'should have no accessibility violations on create task form' do
    # given
    step.navigation.open_main_page
    step.tasks.open_create_task_form

    # when
    results = step.accessibility.analyze

    # then
    validate.accessibility.has_no_violations(results)
  end
end
